core/drivers/pl022_spi.c

```diff
@@ -273,10 +265,11 @@ static void pl022_sanity_check(struct pl022_data *pd)
 {
 	assert(pd);
 	assert(pd->chip.ops);
-	assert(pd->gpio);
-	assert(pd->gpio->ops);
-	assert(pd->base);
-	assert(pd->cs_gpio_base);
+	/* pd->cs_data.cs_cb CAN be NULL so no assert */
+	if (!pd->b_cs_callback) {
```

Comment:
@jforissier Because registering NULL as `cs_data.cs_cb` is a valid use case, so there's no way to know if the caller wants to use `cs_data.cs_cb` or `gpio_data` just by checking `if (cs_data.cs_cb)`, so need a flag to indicate that.

@etienne-lms As above comment in `spi_test.c`, there are 3 use cases:
i. `b_cs_callback` is false, then `pl022_control_cs` will need `cs_data.gpio_data` info to control CS.
ii. `b_cs_callback` is true and `cs_cb` is not NULL, then `pl022_control_cs` will just call `cs_data.cs_cb` to control CS.
iii. `b_cs_callback` is true and `cs_cb` is NULL, then `pl022_control_cs` will do nothing, and let caller control CS manually however they want.

So for `cs_data`, it can only be either `gpio_data` or `cs_cb`, which is the reason an union is used. As in reply to Jerome above, `b_cs_callback` is used to distinguish between the two, because `cs_cb` alone cannot be used to know what caller wants due to use case of allowing registration of NULL as cb.